staging/src/k8s.io/kubectl/pkg/cmd/wait/wait_test.go

```diff
@@ -80,6 +95,17 @@ func addCondition(in *unstructured.Unstructured, name, status string) *unstructu
 	return in
 }
 
+func addConditionWithObservedGeneration(in *unstructured.Unstructured, name, status string, observedGeneration int64) *unstructured.Unstructured {
```

Comment:
Seems like this code change would also need a test for when observedGeneration is on status.

Reply:
Correct I chose to not test all combinations of wait. I figured testing one would be sufficient, as the function to pick observedGeneration from condition or status was fairly basic.

I can add a test case for the following function
https://github.com/kubernetes/kubernetes/blob/44aff9917566885ab5d8cc712a3330bac1923214/staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go#L478-L485

Would you prefer for me to test the function getObservedGeneration directly, or test waiting on .status.observedGeneration to change. Or not test either as it is straight forward.

Reply:
I added tests for .status.observedGeneration changing